src/mixins/filing-template-mixin.ts

```diff
@@ -589,7 +590,7 @@ export default class FilingTemplateMixin extends DateMixin {
     ))
 
     // store current Business Contact
-    this.setBusinessContact({ ...entitySnapshot.authInfo.contact })
+    this.setBusinessContact(filing.correction.contactPoint)
```

Comment:
The other parsing functions in here get the business contact from the entity snapshot, which I think is OK here, too.
Basically, don't restore the business contact from the contact point object. It's special in that you always want to look at the latest data (not from a draft filing).

Reply:
Also this code might fail on the initial correction draft since it probably doesn't contain a contact point object.

Reply:
There is this problem for contactPoint. We need to send the contactPoint json to `AuthServices.updateContactInfo` API when we update it in the component, which also updates the entitysnapshot. Then it can't tell whether contactPoint is changed or not. So I use `filing.correction` as the reference here.

Reply:
I can add a condition to use entitysnapshot if it doesn't contain a contact point object, but this contact point is mandatory according to schema. I'm working on it.

Reply:
I think business contact needs to be fetched from snapshot every time -- never from a draft. Otherwise someone could have updated the business contact separately and then this info would be wrong.

PS Watch you don't confuse business contact (object stored in auth db) and contact point (object in the correction filing).

What I noticed here is that this filing handles things differently from the other filings, and that's unexpected.

Reply:
Okay. I will load it from the entity snapshot, but there will be a minor problem when it comes to saving the draft (unless we don't include `this.isCorrectionFiling` in `onContactPointChange()` in `BusinessContactInfo.vue`). This is issue and why I wanted to use the filing.

Reply:
What's the problem with saving the draft?

Reply:
Changing it back. Sorry for the confusion.